rsconnect/pyproject.py

```diff
@@ -20,18 +21,38 @@ def lookup_metadata_file(directory: typing.Union[str, pathlib.Path]) -> typing.L
 
     The returned value is either a list of tuples [(filename, path)] or
     an empty list [] if no metadata file was found.
+
+    The metadata files are returned in the priority they should be processed
+    to determine the python version requirements.
     """
     directory = pathlib.Path(directory)
 
     def _generate():
-        for filename in ("pyproject.toml", "setup.cfg", ".python-version"):
+        for filename in (".python-version", "pyproject.toml", "setup.cfg"):
             path = directory / filename
             if path.is_file():
                 yield (filename, path)
 
     return list(_generate())
 
 
+def get_python_requires_parser(
+    metadata_file: pathlib.Path,
+) -> typing.Optional[typing.Callable[[pathlib.Path], typing.Optional[str]]]:
+    """Given the metadata file, return the appropriate parser function.
+
+    The returned function takes a pathlib.Path and returns the parsed value.
+    """
+    if metadata_file.name == "pyproject.toml":
+        return parse_pyproject_python_requires
+    elif metadata_file.name == "setup.cfg":
+        return parse_setupcfg_python_requires
+    elif metadata_file.name == ".python-version":
+        return parse_pyversion_python_requires
+    else:
+        return None
```

Comment:
This could be done with a lookup table, but given the options aren't many I preferred to be explicit and avoid dynamic lookups with are harder to follow when debugging.